designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ProcessesResources.scala

```diff
@@ -99,10 +100,14 @@ class ProcessesResources(
             complete {
               // To not overload engine, for list of processes we provide statuses that can be cached
               implicit val freshnessPolicy: DataFreshnessPolicy = DataFreshnessPolicy.CanBeCached
-              processRepository
-                .fetchProcessesDetails[Unit](query.toRepositoryQuery)
-                .flatMap(_.map(enrichDetailsWithProcessState[Unit]).sequence)
-                .toBasicProcess
+              (processRepository
+                .fetchProcessesDetails[Unit](
+                  query.toRepositoryQuery
+                ) zip deploymentService.getProcessesInProgress).flatMap { case (processesList, processesInProgress) =>
+                processesList
+                  .map(enrichDetailsWithProcessState[Unit](_, Some(processesInProgress)))
```

Comment:
Let's add a variant of method with `Map` instead of `Option`, and keep this `Option` as an internal detail that is not published in API of this class

Reply:
I've moved logic to DeploymentService so I'm guessing this one is not applicable anymore